
Fix to use load human-readable error strings and to use
thread-safe error retrieval interface.

--- httpd-2.0.46/modules/ssl/ssl_engine_log.c.sslerr
+++ httpd-2.0.46/modules/ssl/ssl_engine_log.c
@@ -118,8 +118,8 @@
     unsigned long e;
 
     while ((e = ERR_get_error())) {
-        char *err, *annotation;
-        err = ERR_error_string(e, NULL);
+        char err[256], *annotation;
+        ERR_error_string_n(e, err, sizeof err);
         annotation = ssl_log_annotation(err);
 
         if (annotation) {
--- httpd-2.0.46/modules/ssl/ssl_engine_init.c.sslerr
+++ httpd-2.0.46/modules/ssl/ssl_engine_init.c
@@ -1260,7 +1260,6 @@
     /*
      * Try to kill the internals of the SSL library.
      */
-    ERR_free_strings();
     ERR_remove_state(0);
     EVP_cleanup();
 
